native/membrane_video_compositor_common/src/elixir_transfer.rs

```diff
@@ -0,0 +1,108 @@
+use std::{any::Any, sync::Arc};
+
+use crate::plugins::{layout::UntypedLayout, transformation::UntypedTransformation};
+
+pub struct ElixirTypePacket {
+    pointer: (usize, usize),
+}
+
+impl ElixirTypePacket {
+    pub fn encode<T: 'static>(payload: T) -> ElixirTypePacket {
+        let payload: Arc<dyn Any> = Arc::new(payload);
+        let pointer: (usize, usize) = unsafe { std::mem::transmute(payload) };
+
+        ElixirTypePacket { pointer }
+    }
+
+    /// # Safety
+    /// The caller must ensure that all of the following conditions are met:
+    ///  - The packet must have been encoded in the same address space as the call site
+    ///  - The packet must have been encoded in a crate compiled in the same compilation as the call site
+    ///  - The packet must not have been cloned or duplicated in any way in elixir
+    pub unsafe fn decode(self) -> Arc<dyn Any> {
+        let payload: Arc<dyn Any> = unsafe { std::mem::transmute(self.pointer) };
+
+        payload
+    }
+}
+
+impl<'a> rustler::Decoder<'a> for ElixirTypePacket {
+    fn decode(term: rustler::Term<'a>) -> rustler::NifResult<Self> {
+        let pointer: (usize, usize) = rustler::Decoder::decode(term)?;
+        Ok(Self { pointer })
+    }
+}
+
+impl rustler::Encoder for ElixirTypePacket {
+    fn encode<'a>(&self, env: rustler::Env<'a>) -> rustler::Term<'a> {
+        self.pointer.encode(env)
```

Comment:
TBH I have some trust issues with that one :DD 
How is that supposed to work, when `pointer` is `(usize, usize)`?

Reply:
the `rustler::Encoder` trait is automatically implemented by rustler for all tuples with lengths <= 6 provided that the types of all members of those tuples also implement `rustler::Encoder`. Since `usize` implements `rustler::Encoder`, tuples containing `usize`s also do.